pcp/PCPDynamicMeter.c

```diff
@@ -0,0 +1,370 @@
+   PCPDynamicMeter_scanDir(meters, path);
+   free(path);
+
+   /* check the working directory, as a final option */
+   char cwd[PATH_MAX];
+   if (getcwd(cwd, sizeof(cwd)) != NULL) {
+      path = String_cat(cwd, "/pcp/meters/");
+      PCPDynamicMeter_scanDir(meters, path);
+      free(path);
+   }
+}
+
+void PCPDynamicMeter_enable(PCPDynamicMeter* this) {
+   for (unsigned int i = 0; i < this->totalMetrics; i++)
+      Metric_enable(this->metrics[i].id, true);
+}
+
+void PCPDynamicMeter_updateValues(PCPDynamicMeter* this, Meter* meter) {
+   char* buffer = meter->txtBuffer;
+   size_t size = sizeof(meter->txtBuffer);
+   size_t bytes = 0;
+
+   for (unsigned int i = 0; i < this->totalMetrics; i++) {
+      if (i > 0 && bytes < size - 1)
+         buffer[bytes++] = '/';  /* separator */
+
+      PCPDynamicMetric* metric = &this->metrics[i];
+      const pmDesc* desc = Metric_desc(metric->id);
+      pmAtomValue atom;
+
+      if (!Metric_values(metric->id, &atom, 1, desc->type)) {
+         bytes--; /* clear the separator */
+         continue;
+      }
+      /* TODO: pretty-print the values - pmConvScale, etc */
+      switch (desc->type) {
+         case PM_TYPE_STRING:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%s", atom.cp);
+            free(atom.cp);
+            break;
+         case PM_TYPE_32:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%d", atom.l);
+            break;
+         case PM_TYPE_U32:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%u", atom.ul);
+            break;
+         case PM_TYPE_64:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%lld", (long long) atom.ll);
+            break;
+         case PM_TYPE_U64:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%llu", (unsigned long long) atom.ull);
+            break;
+         case PM_TYPE_FLOAT:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%f", (double) atom.f);
+            break;
+         case PM_TYPE_DOUBLE:
+            bytes += xSnprintf(buffer + bytes, size - bytes, "%f", atom.d);
+            break;
+         default:
+            break;
+      }
+   }
+   if (!bytes)
+      xSnprintf(buffer, size, "no data");
+}
+
+void PCPDynamicMeter_display(PCPDynamicMeter* this, ATTR_UNUSED const Meter* meter, RichString* out) {
+   int nodata = 1;
+
+   for (unsigned int i = 0; i < this->totalMetrics; i++) {
+      PCPDynamicMetric* metric = &this->metrics[i];
+      const pmDesc* desc = Metric_desc(metric->id);
+      pmAtomValue atom;
+      char buffer[64];
+      int len;
+
+      if (!Metric_values(metric->id, &atom, 1, desc->type))
+         continue;
+      nodata = 0;
+
+      if (i > 0)
+         RichString_appendnAscii(out, CRT_colors[metric->color], " ", 1);
+
+      if (metric->label) {
+         len = xSnprintf(buffer, sizeof(buffer), "%s ", metric->label);
+         RichString_appendnAscii(out, CRT_colors[metric->color], buffer, len);
```

Comment:
I think the color here should be fixed to match the rest of the htop theme, I suggest this instead:
```
RichString_appendnAscii(out, CRT_colors[METER_TEXT], buffer, len);
```

Reply:
Yep, agreed - will fix up, thanks @smalinux